### DISCUSSION TITLE
Quoted object names

### DISCUSSION THREAD

**Discussion by thiemo-taken** [pgmodeler/pgmodeler#1878] [General]
Hi

I noticed that some object names are generated in quoted form even though quotes were not given in the definition of the object. Is there a specific reason for that? I believe objects not purely in lower case get quoted. As I am not aware of the SQL standard requiring so, and much more, as far as I know, PostgreSQL converts all unquoted names to lower case. I quite feel that there is an unnecessary restriction in place.

I would appreciate some enlightment.

Cheers

Thiemo

**Comment by rkhaotix** [comment]
Hi @thiemo-taken 

The name quoting mechanism in pgModeler follows almost all the rules defined by PostgreSQL, see here:
https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS

**Comment by thiemo-taken** [comment]
Hi @rkhaotix 

Thanks for getting at this.

I am afraid, I do not agree with you. The section mentioned by you states:
![imagen](https://github.com/pgmodeler/pgmodeler/assets/35094030/a0fdfbc8-d386-4837-825d-6ef71577461c)

But pgModeler's
![imagen](https://github.com/pgmodeler/pgmodeler/assets/35094030/f441b652-5405-4600-8d86-db1adfdfdbd7)
results in
![imagen](https://github.com/pgmodeler/pgmodeler/assets/35094030/6476d071-43e2-4376-9748-1afc011046f2)
which in turn get generated into
![imagen](https://github.com/pgmodeler/pgmodeler/assets/35094030/de2e8bde-410e-4ebd-ae1c-aa8850fe8abf)

I see topics there:

1. If a rule or schema, maybe more types, is created not purely in lower case, it automatically gets quoted, already in the internal model visible, if I can take the dbm file to be 1:1 to the internal model. The SQL generation is consistent then. (blue squares in the pictures).
2. All columns, maybe more identifier types, are fine in the internal model, again if it equals dbm file, but the SQL generation then quotes them (yellow patches).

I have to admit, I did not test the very latest pgModeler version as I do not have access to it. My latest is 1.0.1. It might be different in later ones.
![imagen](https://github.com/pgmodeler/pgmodeler/assets/35094030/dfe8a723-0c25-4d0a-9226-4768da181ec0)

Long live Asterix!

Thiemo

**Comment by rkhaotix** [comment]
It's ok to disagree. That's why I said "almost all the rules". :)

pgModeler historically (and by request from users in the past) automatically double-quotes any object name that isn't purely lower case. This is also a consent among PostgreSQL community. If you will use upper case latters, make sure to always type the object names between doubleq uotes to avoid confusion. That's why they recommend to always use lower case names :)

Unfortunately, I can't say that to make the automatic double-quotes optional via setting is easy, because it isn't, there are too many things that depends on it. :(

**Comment by thiemo-taken** [comment]
I see, a deliberate break with the working of PostgreSQL then. I know of the recommendation of PostgreSQL to use lower case identifiers and upper case keywords. To me, it makes rather more sense to write the identifiers upper, and the keywords lower case, sort of counter-weight to the syntax highlighted colouring of the keywords. But that is just a personal preference.
I have to admit, that I am a bit disappointed about the removal of the freedom to choose my way, but pgModeler is still a great tool.

**Reply by rkhaotix** [reply]
Don't worry, I'll dig into the internals to see if I can easily change that behavior. :)
